### PR TITLE
make real_registered_proxy_url take the correct port

### PR DESCRIPTION
**Pull request by cristifalcas**
make real_registered_proxy_url take the correct port

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -450,6 +450,12 @@
   }
   validate_absolute_path($puppet_cache_location)
 
+  if $registered_proxy_url {
```
Comment (ekohl) [inline manifests/init.pp]: Would you mind using ```pick()``` here?

File: spec/classes/foreman_proxy__register__spec.rb [hunk 2]
```diff
@@ -49,6 +49,33 @@
         end
       end
 
+      describe 'with ssl_port overrides' do
+        let :pre_condition do
+          "class {'foreman_proxy':
+        register_in_foreman   => true,
```
Comment (ekohl) [inline spec/classes/foreman_proxy__register__spec.rb]: The indenting appears to be off here.

File: spec/classes/foreman_proxy__register__spec.rb [hunk 3]
```diff
@@ -49,6 +49,33 @@
...
+        foreman_base_url      => 'my_base',
+        registered_name       => 'my_proxy',
+        registered_proxy_url  => 'my_url',
+        oauth_consumer_key    => 'key',
+        oauth_consumer_secret => 'secret',
+        oauth_effective_user  => 'smartproxy',
+	ssl_port              => '1234',
```
Comment (ekohl) [inline spec/classes/foreman_proxy__register__spec.rb]: Can you replace the tab with spaces here?

File: spec/classes/foreman_proxy__register__spec.rb (hunk 2 above)
Comment (ekohl) [inline spec/classes/foreman_proxy__register__spec.rb]: I now see this was already present in the other test.

**Comment by ekohl** [comment]
:+1: code wise

**Comment by ehelms** [comment]
Thanks for tackling this @cristifalcas !

**Comment by cristifalcas** [comment]
done

**Comment by mmoll** [comment]
@cristifalcas tests are red

**Comment by ehelms** [comment]
Should https://github.com/theforeman/puppet-foreman_proxy/pull/211/files#diff-60ae41fd0a31977447947f59940ee9a4R378 be changed to undef either in the init.pp or params.pp so that it is clearer and an obvious value?

File: manifests/params.pp [hunk 4]
```diff
@@ -246,8 +246,6 @@
   $foreman_base_url = "https://${lower_fqdn}"
   # Name the proxy should be registered with
   $registered_name = $::fqdn
-  # Proxy URL to be registered
-  $registered_proxy_url = "https://${registered_name}:${ssl_port}"
```
Comment (ekohl) [inline manifests/params.pp]: @ehelms is correct, this should be undef.

**Comment by cristifalcas** [comment]
fixed

**Comment by mmoll** [comment]
:+1:

**Comment by ekohl** [comment]
Thanks! I was too soon and didn't check if tests passed. I'll check back and fix them if needed.

**Comment by ehelms** [comment]
Thanks again for the expediency on this @cristifalcas !

**Comment by mmoll** [comment]
Tests were green when I commented, I canceled the (new) PR tests and we'll see what the push test will bring...

**Pull request merged**